includes/class-wp-document-revisions-admin.php

```diff
@@ -884,6 +890,83 @@ public function profile_update_cb() {
 		}
 	}
 
+	/**
+	 * Ensures that any system limit on revisions does not apply to documents
+	 *
+	 * @since 3.2.2
+	 *
+	 * @param int     $num  default value for the number of revisions for the post_type
+	 * @param WP_Post $post current post
+	 */
+	public function manage_document_revisions( $num, $post ) {
+
+		if ( ! is_object( $post ) || 'document' !== $post->post_type ) {
```

Comment:
Would it make sense to use `verify_post_type` here?

Reply:
Agreed